src/threading/thread.cpp

```diff
@@ -156,21 +155,30 @@ bool Thread::stop()
 }
 
 
-void Thread::wait()
+bool Thread::wait()
 {
-	if (!m_running)
-		return;
+	MutexAutoLock lock(m_mutex);
+
+	if (!m_joinable)
```

Comment:
As we have the mutex now, we can keep the name for the variable to be m_running, right?